gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/IncrementalTest.kt

```diff
@@ -0,0 +1,545 @@
+
+          interface TypeA
+          interface TypeB
+
+          class Consumer @Inject constructor(
+            private val dep: $superType
+          )
+        """.trimIndent()
+
+      rootProject {
+        gradlePropertiesFile(
+          """
+          com.squareup.anvil.trackSourceFiles=true
+          com.squareup.anvil.addOptionalAnnotations=true
+          """.trimIndent(),
+        )
+
+        settingsFileAsFile.appendText(
+          """
+
+          include(":lib")
+          include(":app")
+          """.trimIndent(),
+        )
+
+        project("lib") {
+          buildFile {
+            plugins {
+              kotlin("jvm")
+              id("com.squareup.anvil")
+            }
+            anvil {
+              generateDaggerFactories.set(true)
+            }
+            dependencies {
+              implementation(libs.dagger2.annotations)
+            }
+          }
+          otherClassPath.createSafely(otherClassContent("TypeA"))
+        }
+
+        project("app") {
+
+          buildFile {
+            plugins {
+              kotlin("jvm")
+              kotlin("kapt")
+              id("com.squareup.anvil")
+            }
+            dependencies {
+              compileOnly(libs.dagger2.annotations)
+              implementation(project(":lib"))
+              kapt(libs.dagger2.compiler)
+            }
+          }
+
+          dir("src/main/java") {
+            kotlinFile(
+              "com/squareup/test/app/AppComponent.kt",
+              """
+              package com.squareup.test.app
+
+              import com.squareup.anvil.annotations.MergeComponent
+              import javax.inject.Singleton
+              import com.squareup.anvil.annotations.optional.SingleIn
+
+              @SingleIn(Any::class)
+              @MergeComponent(Any::class)
+              interface AppComponent
+              """.trimIndent(),
+            )
+          }
+        }
+      }
+
+      shouldSucceed(":app:compileJava")
+
+      val componentModuleFile = rootProject.path.resolve("app")
+        .anvilMainGenerated
+        .resolve("anvil/module/com/squareup/test/app/AppComponent.kt")
+
+      componentModuleFile.shouldExist()
+      componentModuleFile.readText() shouldContain """
+        @Module
+        @ContributesTo(Any::class)
+        public abstract class AppComponentAnvilModule {
+          @Binds
+          public abstract fun bindTypeA(otherClass: OtherClass): TypeA
+        }
+      """.trimIndent()
+
+      otherClassPath.writeText(otherClassContent("TypeB"))
+
+      shouldSucceed(":app:compileJava") {
+        task(":app:compileKotlin")?.outcome shouldBe TaskOutcome.SUCCESS
+      }
+
+      componentModuleFile.shouldExist()
+      componentModuleFile.readText() shouldContain """
+        @Module
+        @ContributesTo(Any::class)
+        public abstract class AppComponentAnvilModule {
+          @Binds
+          public abstract fun bindTypeB(otherClass: OtherClass): TypeB
+        }
```

Comment:
Nice, yea this is the type of content assertion I think we should include in the other test